src/sortStudents.ts

```diff
@@ -1,16 +1,67 @@
-
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  const sortedStudents: Student[] = [...students];
+
+  type Callback = (x: number, y: number) => number;
+
+  const sum: Callback = (x, y) => x + y;
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      if (order === 'asc') {
+        sortedStudents.sort((a: Student, b: Student) => a[sortBy]
+          .localeCompare(b[sortBy]));
+      } else {
+        sortedStudents.sort((a: Student, b: Student) => b[sortBy]
+          .localeCompare(a[sortBy]));
+      }
+      break;
+
+    case SortType.Age:
+    case SortType.Married:
+      if (order === 'asc') {
+        sortedStudents
+          .sort((a: Student, b: Student) => +a[sortBy] - +b[sortBy]);
```

Comment:
Try to use Number() constuctor, it's more obvious and noticeable in code than operator `+`
```suggestion
          .sort((a: Student, b: Student) => Number(a[sortBy]) - +b[sortBy]);
```